src/ui/React/ServerDropdown.tsx

```diff
@@ -35,24 +36,35 @@ export function ServerDropdown(props: IProps): React.ReactElement {
    * Checks if the server should be shown in the dropdown menu, based on the
    * 'serverType' property
    */
-  function isValidServer(s: BaseServer): boolean {
-    const purchased = (s instanceof Server && s.purchasedByPlayer) || s instanceof HacknetServer;
+  function isValidServer(baseServer: BaseServer): boolean {
+    /**
+     * isOwnedServer is true if baseServer is home, private servers, or hacknet servers. Note that home satisfies the
+     * condition: (baseServer instanceof Server && baseServer.purchasedByPlayer).
+     */
+    const isOwnedServer =
+      (baseServer instanceof Server && baseServer.purchasedByPlayer) || baseServer instanceof HacknetServer;
     const type = props.serverType;
     switch (type) {
       case ServerType.All:
         return true;
       case ServerType.Foreign:
-        return s.hostname !== "home" && !purchased && !Player.hasAugmentation(AugmentationName.TheRedPill, true)
-          ? s.hostname !== "w0r1d_d43m0n"
-          : true;
+        // Exclude home, private servers, hacknet servers.
+        if (isOwnedServer) {
+          return false;
+        }
+        // If the player has not installed TRP, exclude WD server.
+        if (!Player.hasAugmentation(AugmentationName.TheRedPill, true)) {
```

Comment:
Optional: Can be simplified to (pseudocode) `return hasAugmentation || hostname !== WorlDaemon`